de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/TextEditActivityExecutor.java

```diff
@@ -0,0 +1,45 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import de.fu_berlin.inf.dpp.activities.TextEditActivity;
+import de.fu_berlin.inf.dpp.server.editor.ServerEditorManager;
+import de.fu_berlin.inf.dpp.session.AbstractActivityConsumer;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+import org.picocontainer.Startable;
+
+/**
+ * Executes TextEditActivities through the {@link ServerEditorManager}.
+ */
+public class TextEditActivityExecutor extends AbstractActivityConsumer implements
+    Startable {
+
+    ISarosSession session;
+    ServerEditorManager editorManager;
+
+    /**
+     * Creates a TextEditActivityExecutor.
+     *
+     * @param session
+     *            the current session
+     * @param editorManager
+     *            the editor manager
+     */
+    public TextEditActivityExecutor(ISarosSession session,
+        ServerEditorManager editorManager) {
+
+        this.session = session;
+        this.editorManager = editorManager;
+    }
+
+    @Override public void start() {
+        session.addActivityConsumer(this, Priority.ACTIVE);
+    }
+
+    @Override public void stop() {
+        session.removeActivityConsumer(this);
+    }
+
+    @Override
+    public void receive(TextEditActivity activity) {
+        editorManager.applyTextEdit(activity);
```

Comment:
much boiler code for actually 1 line of actual logic. And the worst part is that that consumer is a complete different package (I know that there is no common rule where classes should be placed).

Why not move this code directly to the EditorManager ?

Reply:
In that case the `ServerEditorManager` would also need to extend the `AbstractActivityConsumer` or are you talking about changing the core interface?

Either way this is how it is handled in all other plugins and the `EditorManager` is only involved, because ultimately the Editor is responsible for applying the text edit, but you cannot let each editor listen for activities easily. I feel this is the cleanest way to split the code and responsibilities, although this indeed looks like boilerplate.

All you other comments should be resolved, please check again. :)

Reply:
https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/editor/EditorManager.java#L161 

?

Reply:
Okay, you are absolutely right, my apologies.
However this would also involve quite some boilerplate, as I would need to setup an `ISessionLifetimeListener` to add and remove the `IActivityExecutor` (https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/editor/EditorManager.java#L356).

Looks like almost the same amount of code to me and not really worth for just one type of Activity. Especially because I think this version is easier to understand.

If you insist, I will change it, but you approved the PR, so I assume this is not that important, right?